### PR TITLE
Added support of SPL ArrayObject

### PR DESCRIPTION
**Pull request by alexvasi**
I've added support of ArrayObject (from SPL). I know, it's kind of hard to work with PhpArrayObject, but at least serialization works.

### CODE REVIEW

File: php_serialize/common.go [hunk 1]
```diff
@@ -11,6 +11,8 @@ const (
 	TOKEN_OBJECT_SERIALIZED rune = 'C'
 	TOKEN_REFERENCE         rune = 'R'
 	TOKEN_REFERENCE_OBJECT  rune = 'r'
+	TOKEN_ARRAYOBJECT       rune = 'x'
```
Comment (kihamo) [inline php_serialize/common.go]: Please  rename to TOKEN_SPL_ARRAY

File: php_serialize/common.go [hunk 2]
```diff
@@ -11,6 +11,8 @@ const (
 	TOKEN_OBJECT_SERIALIZED rune = 'C'
 	TOKEN_REFERENCE         rune = 'R'
 	TOKEN_REFERENCE_OBJECT  rune = 'r'
+	TOKEN_SPL_ARRAY         rune = 'x'
+	TOKEN_SPL_ARRAY_MEMBERS rune = 'm'
```
Comment (Barberrrry) [inline php_serialize/common.go]: Please, rename it to TOKEM_SPL_ARRAY_MEMBER, because it's token of ONE member.

File: php_serialize/serialize.go [hunk 3]
```diff
@@ -45,6 +45,8 @@ func (self *Serializer) Encode(v PhpValue) (string, error) {
 		value = self.encodeObject(v)
 	case *PhpObjectSerialized:
 		value = self.encodeSerialized(v)
+	case *PhpSplArray:
+		value = self.encodeArrayObject(v)
```
Comment (Barberrrry) [inline php_serialize/serialize.go]: Why didn't you name the method encodeSplArray according to data type?

File: php_serialize/serialize_test.go [hunk 4]
```diff
@@ -414,3 +414,18 @@ func TestEncodeObjectSerializableJSON(t *testing.T) {
...
+	obj := NewPhpSplArray(PhpArray{"foo": 42}, nil)
+
+	data, err := Serialize(obj)
+	if err != nil {
+		t.Errorf("Error while encoding array object: %v\n", err)
+	}
+
+	expected := "x:i:0;a:1:{s:3:\"foo\";i:42;};m:a:0:{}"
+
+	if !strings.Contains(data, expected) {
```
Comment (Barberrrry) [inline php_serialize/serialize_test.go]: Why you don't check equality of strings here?

**Pull request merged**